suzieq/engines/pandas/interfaces.py

```diff
@@ -56,20 +56,15 @@ def get(self, **kwargs):
         if df.empty:
             return df
 
-        if portmode or 'portmode' in columns or '*' in columns:
+        if portmode or any(x in fields
+                           for x in ['vlan', 'vlanList', 'portmode']):
             for x in ['ipAddressList', 'ip6AddressList']:
                 if x in columns or '*' in columns:
                     continue
                 drop_cols.append(x)
             df = self._add_portmode(df, **kwargs)
 
         if vlan or "vlanList" in columns or '*' in columns:
```

Comment:
```suggestion
        if vlan or 'vlanList' in fields:
```
It should do the same thing, but it's more clear

Reply:
Ack